"What's new in Svelte" April newsletter (#6129)
sveltejs/svelte: site/content/blog/2021-04-01-whats-new-in-svelte-april-2021.md

---

> title: What's new in Svelte: March 2021

Reviewer: ```suggestion
title: What's new in Svelte: April 2021
```
😄

description: SvelteKit beta and new way to use slots
author: Daniel Sandoval
authorURL: https://desandoval.net
---

Two projects that have been months (even years) in the making have made their way out into the world. SvelteKit is now in public beta and slotted components are now available in Svelte!

## What's up with SvelteKit?

> SvelteKit - Svelte's meta-framework for building SSR, serverless applications - is now officially in public beta. Expect bugs! Lots more detail in the [latest blog post](https://svelte.dev/blog/sveltekit-beta). Want to know when 1.0 is close? Check out the milestone on [github](https://github.com/sveltejs/kit/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0).

Reviewer 2: ```suggestion
SvelteKit - Svelte's meta-framework for building SSR, serverless applications - is now officially in public beta. Expect bugs! Lots more detail in the [latest blog post](https://svelte.dev/blog/sveltekit-beta). Want to know when 1.0 is close? Check out the milestone on [github](https://github.com/sveltejs/kit/milestone/2).
```
I think that view feels more natural

> SvelteKit - Svelte's meta-framework for building SSR, serverless applications - is now officially in public beta. Expect bugs! Lots more detail in the [latest blog post](https://svelte.dev/blog/sveltekit-beta). Want to know when 1.0 is close? Check out the milestone on [github](https://github.com/sveltejs/kit/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.0).

Reviewer 3: I'm not sure what's meant by "meta" framework. I'd just call it a framework

## New in Svelte & Language Tools

> - Slotted components, including `<svelte:fragment slot="...">` lets component consumers target specific slots with rich content (**Svelte 3.35.0, Language Tools [104.5.0](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.5.0)**, [documentation](https://svelte.dev/docs#slot_name))

Reviewer 4: It might be good to note that this is currently not documented, though there's a PR open to add docs (#6097).

That PR has a pretty good description of the new feature:
> Components can be placed in a named slot using the syntax `<Component slot="name" />`.
In order to place content in a slot without losing the flow layout (ie: without using a wrapper element) you can use the special element `<svelte:fragment>`.

Author: Thanks! Do you think the docs will be merged by the first? If so, what anchor link do you think would make the most sense to point to here? Adding your explanation as a stop-gap just in case :)

Reviewer 4: @dummdidumm just approved the PR a few hours ago, so maybe it will be merged in time?

If it does get merged, the docs on [named slots](https://svelte.dev/docs#slot_name) and [svelte:fragment](https://svelte.dev/docs#svelte_fragment) (not live yet), as well as the new tutorial on [svelte:fragment](https://svelte.dev/tutorial/svelte-fragment) (also not live) would be good to link.

Reviewer 5: It's live now 
https://svelte.dev/tutorial/svelte-fragment 
https://svelte.dev/docs#slot_name
https://svelte.dev/docs#svelte_fragment

- Linked editing now works for HTML in Svelte files (**Language Tools, [104.6.0](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.6.0)**)
- Type definitions `svelte.d.ts` are now resolved in order, allowing library authors to ship type definitions with their svelte components (**Language Tools, [104.7.0](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.7.0)**)

---

## Community Showcase

**Apps & Sites**

- [Nagato](https://nagato.app/) is a task management tool that connects popular time tracking and to-do tools all in one place.
- [type-kana](https://type-kana.cass.moe/setup) is a quiz app to help you learn ひらがな (hiragana) and カタカナ (katakana), the Japanese syllabaries.
- [Pittsburgh Steps](https://pittsburgh-steps.samlearner.com/) is an interactive map of the more than 800 sets of public outdoor stairways in Pittsburgh, PA.
- [Music Mode Wheels](https://tobx.github.io/music-mode-wheels/) is a website that displays music modes as interactive wheels.
- [Critical Notes](https://www.critical-notes.com/) helps game masters and players keep track of their roleplaying games campaigns and adventures.

> - [Svelte Game of Life](https://github.com/alanrsoares/svelte-game-of-life) is an educational implementation of Conway's Game of Life in Typescript + Svelte

Reviewer 4: Consistency with other mentions of TypeScript:

```suggestion
- [Svelte Game of Life](https://github.com/alanrsoares/svelte-game-of-life) is an educational implementation of Conway's Game of Life in TypeScript + Svelte
```

- [foxql](https://github.com/foxql) is a peer to peer full text search engine that runs on your browser.

**Demos, Libraries, Tools & Components**

- [svelte-nodegui](https://github.com/nodegui/svelte-nodegui) is a way to build performant, native and cross-platform desktop applications with Node.js and Svelte
- [Svelte Story Format](https://www.npmjs.com/package/@storybook/addon-svelte-csf) Allows to write your "stories" in Storybook using the Svelte syntax. More info in the [Storybook blog](https://storybook.js.org/blog/storybook-for-svelte/)
- [SelectMadu](https://github.com/pavish/select-madu) is a replacement for the select menu, with support for searching, multiple selections, async data loading and more.
- [Svelte Checklist](https://www.npmjs.com/package/svelte-checklist) is a customizable CheckList built with Svelte.

> - [Suspense for Svelte](https://www.npmjs.com/package/@jamcart/suspense) is a Svelte component that implements the core idea of React's <Suspense>.

Reviewer 5: Because Suspense is in tags, it isn't rendered

- [MiniRx](https://spierala.github.io/mini-rx-store/) is a RxJS Redux Store that works with Svelte and TypeScript
- [svelte-formly](https://github.com/arabdevelop/svelte-formly) generates dynamic forms for Svelte and Sapper
- [7ty](https://www.npmjs.com/package/@jamcart/7ty) is a static site generator that uses Svelte, supports partial hydration of components, and uses file based routing resembling Sapper and 11ty.

**Want to contribute your own component?** Submit a [Component](https://sveltesociety.dev/components) to the Svelte Society site by making [a PR to this file](https://github.com/svelte-society/sveltesociety.dev/blob/master/src/pages/components/components.json).

**Starters**

- [sveltekit-electron](https://github.com/FractalHQ/sveltekit-electron) is a starter kit for Electron using SvelteKit
- [sveltekit-tailwindcss-external-api](https://github.com/acidlake/sveltekit-tailwindcss-external-api) is everything you need to build a Svelte project with TailwindCSS and an external API, powered by create-svelte.

> - [Sapper Netlify](https://www.npmjs.com/package/sapper-netlify) is a Sapper project that can run on a netlify function.

Reviewer 6: I think my opinion on that is unfortunately it is neither a good time to start a new project in Sapper nor a good time to start a new project in SvelteKit, although one of those it will be a good time for fairly soon, and the other it will not.

Besides the bugs that are being quickly ironed out, I think the only reason to use Sapper at this point would be if you didn't want to wait for the differential/legacy build stuff in Kit to settle.

> - [Sapper Netlify](https://www.npmjs.com/package/sapper-netlify) is a Sapper project that can run on a netlify function.

Reviewer 4: Assuming this stays in the newsletter, I think Netlify should be capitalized
```suggestion
- [Sapper Netlify](https://www.npmjs.com/package/sapper-netlify) is a Sapper project that can run on a Netlify function.
```

**Learning Resources**
- [How to Build a Website with Svelte and SvelteKit](https://prismic.io/blog/svelte-sveltekit-tutorial) is a step-by-step tutorial walking through the new SvelteKit setup.

> - [A Svelte store for prefers-reduced-motion](https://geoffrich.net/posts/svelte-prefers-reduced-motion-store/) demonstrates how to make a custom Svelte store whose value will indicate whether the user has requested reduced motion #a11y

Reviewer 5: The hashtag feels strange here since there were no usages if hashtags before

---
title: What's new in Svelte: April 2021
description: SvelteKit beta and new way to use slots
author: Daniel Sandoval
authorURL: https://desandoval.net
---

Two projects that have been months (even years) in the making have made their way out into the world. SvelteKit is now in public beta and slotted components are now available in Svelte!

## What's up with SvelteKit?

> SvelteKit - Svelte's framework for building SSR, serverless applications - is now officially in public beta. Expect bugs! Lots more detail in the [latest blog post](https://svelte.dev/blog/sveltekit-beta). Want to know when 1.0 is close? Check out the milestone on [github](https://github.com/sveltejs/kit/milestone/2).

Reviewer 7: ```suggestion
[SvelteKit](https://kit.svelte.dev/) - Svelte's framework for building SSR, serverless applications - is now officially in public beta. Expect bugs! Lots more detail in the [latest blog post](https://svelte.dev/blog/sveltekit-beta). Want to know when 1.0 is close? Check out the milestone on [github](https://github.com/sveltejs/kit/milestone/2).
```
Worth linking to the site too

Want to learn more about how to get started, what's different compared to Sapper, new features and migration paths? Check out this week's [episode of Svelte Radio](https://www.svelteradio.com/episodes/svelte-kit-public-beta) for a deep dive with Antony, Kev and Swyx.

## New in Svelte & Language Tools
- Slotted components, including `<svelte:fragment slot="...">` lets component consumers target specific slots with rich content (**Svelte 3.35.0, Language Tools [104.5.0](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.5.0)**, check out the [docs](https://svelte.dev/docs#svelte_fragment) and the [tutorial](https://svelte.dev/tutorial/svelte-fragment))
- Linked editing now works for HTML in Svelte files (**Language Tools, [104.6.0](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.6.0)**)
- Type definitions `svelte.d.ts` are now resolved in order, allowing library authors to ship type definitions with their svelte components (**Language Tools, [104.7.0](https://github.com/sveltejs/language-tools/releases/tag/extensions-104.7.0)**)

> 

Reviewer 8: - [vite-plugin-svelte](https://github.com/sveltejs/vite-plugin-svelte) is available for general use of svelte in vite and `npm init @vitejs/app` now includes svelte options
